### PR TITLE
Allow for different DB paths for immutable and volatile data

### PR DESCRIPTION
**Pull request by jasagredo**
Part of IntersectMBO/cardano-node#5925, only pending integration into the node.

### CODE REVIEW

**Comment by jasagredo** [comment]
Adding the no-changelog label to omit the changes in the unstable-cardano-tools

File: ouroboros-consensus-diffusion/src/ouroboros-consensus-diffusion/Ouroboros/Consensus/Node.hs [hunk 1]
```diff
@@ -284,6 +289,21 @@ data LowLevelRunNodeArgs m addrNTN addrNTC versionDataNTN versionDataNTC blk
     , llrnPublicPeerSelectionStateVar :: StrictSTM.StrictTVar m (Diffusion.PublicPeerSelectionState addrNTN)
     }
 
+data NodeDatabasePaths =
+    OnePathForAllDbs FilePath
+  | MultipleDbPaths
+      FilePath -- ^ Immutable path, usually pointing to a non-necessarily
+               -- performant volume
+      FilePath -- ^ Volatile path, usually pointing to a performant volume
```
Comment (amesgen) [inline ouroboros-consensus-diffusion/src/ouroboros-consensus-diffusion/Ouroboros/Consensus/Node.hs:297]: I think it makes to make it clear that this is not just for the VolatileDB, but for everything apart from the ImmutableDB. Maybe even call it "non-immutable"? Or go even further and have a record type with different `FilePath`s for each component?

**Review by amesgen — APPROVED** [review]
LGTM, only a few non-crucial comments on documentation/naming

File: ouroboros-consensus-diffusion/src/ouroboros-consensus-diffusion/Ouroboros/Consensus/Node.hs [hunk 2]
```diff
@@ -220,9 +221,13 @@ data LowLevelRunNodeArgs m addrNTN addrNTC versionDataNTN versionDataNTC blk
...
     , llrnChainDbArgsDefaults :: Incomplete ChainDbArgs m blk
 
-      -- | File-system on which the directories for the different databases will
+      -- | File-system on which the directory for the immutable database will
       -- be created.
-    , llrnMkHasFS :: ChainDB.RelativeMountPoint -> SomeHasFS m
+    , llrnMkImmutableHasFS :: ChainDB.RelativeMountPoint -> SomeHasFS m
+
+      -- | File-system on which the directories for the volatile databases will
+      -- be created.
```
[the remark this answers is not in the document]
Comment (jasagredo) [reply ouroboros-consensus-diffusion/src/ouroboros-consensus-diffusion/Ouroboros/Consensus/Node.hs]: Clarified

File: ouroboros-consensus-diffusion/src/ouroboros-consensus-diffusion/Ouroboros/Consensus/Node.hs (hunk 1 above)
Comment (jasagredo) [reply ouroboros-consensus-diffusion/src/ouroboros-consensus-diffusion/Ouroboros/Consensus/Node.hs]: I called it `nonImmutablePath`, I think it is enough this way.

File: ouroboros-consensus-diffusion/src/ouroboros-consensus-diffusion/Ouroboros/Consensus/Node.hs [hunk 3]
```diff
@@ -284,6 +289,21 @@ data LowLevelRunNodeArgs m addrNTN addrNTC versionDataNTN versionDataNTC blk
...
+               -- performant volume
+      FilePath -- ^ Volatile path, usually pointing to a performant volume
+
+immutableDbPath :: NodeDatabasePaths -> FilePath
+immutableDbPath (OnePathForAllDbs f)    = f
+immutableDbPath (MultipleDbPaths imm _) = imm
+
+volatileDbPath :: NodeDatabasePaths -> FilePath
+volatileDbPath (OnePathForAllDbs f)    = f
+volatileDbPath (MultipleDbPaths _ vol) = vol
```
[the remark this answers is not in the document]
Comment (jasagredo) [reply ouroboros-consensus-diffusion/src/ouroboros-consensus-diffusion/Ouroboros/Consensus/Node.hs]: Clarified

**Pull request merged**